index.js

```diff
@@ -861,6 +862,19 @@ function _applyFragment (fragment) {
         value = value.split(',')
       }
 
+      if (action === 'proposal') {
+        const maxProposalIDLength = 7
+
+        value = value.filter(function (id) {
+          // filter out invalid identifiers.
+          return id.match(/^SE-\d+$/i) && id.length <= maxProposalIDLength
+        }).map(function (id) {
+          // insert missing leading zeros, e.g., 'SE-2' → 'SE-0002'.
+          const numMissingLeadingZeros = maxProposalIDLength - id.length
+          return id.substring(0, 3) + '0'.repeat(numMissingLeadingZeros) + id.substring(3)
+        })
```

Comment:
Could we use [`flatMap`][] and [`padStart`][] as follows?

[`flatMap`]:  <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap>
[`padStart`]: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart>

```suggestion
        value = value.flatMap(function (id) {
          const output = id.match(/^SE-(\d{1,4})$/i)
          if (!output) return []
          return 'SE-' + parseInt(output[1]).toString().padStart(4, '0')
        })
```

Reply:
Sure! 

> [`padStart`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)

Cool, I’m surprised there’s an API for such a seemingly uncommon task. Why the `parseInt` though? I think we can avoid it by replacing the `\d` with `[0-9]`, in case they aren’t equivalent here.